specification/authorization/resource-manager/Microsoft.Authorization/preview/2021-11-16-preview/examples/GetAccessReviewHistoryDefinitionInstances.json

```diff
@@ -0,0 +1,31 @@
+{
+  "parameters": {
+    "api-version": "2021-11-16-preview",
+    "subscriptionId": "129a304b-4aea-4b86-a9f7-ba7e2b23737a",
+    "historyDefinitionId": "44724910-d7a5-4c29-b28f-db73e717165a"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "44724910-d7a5-4c29-b28f-db73e717165a",
+            "name": "44724910-d7a5-4c29-b28f-db73e717165a",
+            "type": "Microsoft.Authorization/accessReviewHistoryInstance",
+            "properties": {
+              "reviewHistoryPeriodStartDateTime": "2021-10-01T00:00:00-07:00",
+              "reviewHistoryPeriodEndDateTime": "2021-12-01T00:00:00-08:00",
+              "displayName": "Hello world name",
+              "status": "Requested",
+              "runDateTime": "2021-12-23T00:27:33.6905189+00:00",
+              "fulfilledDateTime": "2021-12-23T00:28:33.6905189+00:00",
+              "downloadUri": "https://dfermconsolreportusc.blob.core.windows.net/df-erm-reports/Hello world name-22be232e-a93d-42a3-8ac5-313cfd29a0eb.csv?sv=2015-04-05&ss=b&srt=o&sp=rl&st=2021-03-22T19:39:38.0000000Z&se=2021-03-23T19:41:38.0000000Z&spr=https&sig=84rlGCIgU4ToMn%2FFLncBXq95O8a8RsFlwQY1Knl%2Fo%2FI%3D",
```

Comment:
>downloadUri [](http://example.com/codeflow?start=15&length=11)

Is this value always returned / generated? If yes, then maybe you don't need to model a POST API -> `generateDownloadUri` and you are simply best with updating your POST to GET method. Do you generate this value on every single instance item every time you return a list?

Reply:
No the uri is only generated when this action is called but we do return the URI as customers will want to come back and re-download the results again at a later time.